dotnet/samples/KernelSyntaxExamples/Example76_Filters.cs

```diff
@@ -212,6 +204,58 @@ public async Task OnFunctionInvocationAsync(FunctionInvocationContext context, F
         }
     }
 
+    /// <summary>Shows syntax for prompt filter.</summary>
+    private sealed class PromptFilterExample : IPromptFilter
+    {
+        public async Task OnPromptRenderingAsync(PromptRenderingContext context, Func<PromptRenderingContext, Task> next)
+        {
+            // Example: get function information
+            var functionName = context.Function.Name;
+
+            await next(context);
+
+            // Example: override rendered prompt before sending it to AI
+            context.RenderedPrompt = "Safe prompt";
+        }
+    }
+
+    /// <summary>Shows syntax for function call filter.</summary>
+    private sealed class FunctionCallFilter(ITestOutputHelper output) : IFunctionCallFilter
+    {
+        private readonly ITestOutputHelper _output = output;
+
+        public async Task OnFunctionCallInvocationAsync(FunctionCallInvocationContext context, Func<FunctionCallInvocationContext, Task> next)
+        {
+            // Example: get function information
+            var functionName = context.Function.Name;
+
+            // Example: get function call iteration information
+            this._output.WriteLine($"Function call iteration: {context.FunctionCallIteration + 1} out of {context.FunctionCallCount}");
+
+            // Example: get request iteration
+            this._output.WriteLine($"Request iteration: {context.RequestIteration + 1}");
+
+            // Calling next filter in pipeline or function itself.
+            // By skipping this call, next filters and function won't be invoked, and function call loop will proceed to the next function.
+            await next(context);
+
+            // Example: get function result
+            var result = context.Result;
+
+            // Example: override function result value
+            context.Result = new FunctionResult(context.Result, "Result from function call filter");
+
+            // Example: stop further function calling. but proceed with request iteration
+            context.Action = FunctionCallAction.StopFunctionCallIteration;
+
+            // Example: call remaining functions, but stop request iteration
+            context.Action = FunctionCallAction.StopRequestIteration;
+
+            // Example: stop function calling and request iterations, return immediately
+            context.Action = FunctionCallAction.StopRequestIteration | FunctionCallAction.StopFunctionCallIteration;
```

Comment:
Should we define something like `FunctionCallAction.ReturnResult` to aid discoverability?

Reply:
I simplified the implementation and removed Action enumeration completely in favour of simple boolean `Cancel` property.